docs/adrs/flannel-options.md

```diff
@@ -0,0 +1,110 @@
+| `--public-ipv6 (string)`| hardcoded | `--external-ip` when it is of IPv6 nature if k3s server flag `--flannel-external-ip` is true |
+
+** Does not provide much value as flannel in k3s can configure the interface per node
+
+Regarding subnet and backend configuration:
+
+`Network` and `IPv6Network` are set by reading the assigned podCIDR for the Kubernetes node.
+`EnableIPv6` is set based on what the user passed as k3s server `--cluster-cidr`.
+`Backend` uses vxlan as default, although this can be changed using k3s server flag `--flannel-backend`
+
+Something important to note is that k3s allows the user to override the whole subnet and backend configuration, by using the k3s agent flag `--flannel-conf` and even the flannel CNI configuration file, by using the k3s agent flag `--flanel-cni-conf`. Overriding the whole subnet and backend configuration is useful if user wants to add specific backend configurations, e.g. VNI for vxlan. Users can currently do this by using `<=option1=val1,option2=val2>` when selecting the backend but it seldom used and complicates things, thus it was deprecated in favor of `--flannel-cni-conf`.
+
+To wrap up the context, k3s includes the following flannel options:
+
+| Agent Flag | Type | Description |
+| --- | --- | --- |
+| `--flannel-iface` | string | Overrides the default flannel interface. This interface is used to forward encapsulated traffic in inter-node communication. It matches the flannel flag `--iface` |
+| `--flannel-conf` | string | Path that points to a file containing the flannel subnet&backend config |
+| `--flannel-cni-conf` | string | Path that point to a flannel CNI config file. This config sets CNI specific configuration for that node |
+
+| Server Flag | Type | Description |
+| --- | --- | --- |
+| `--flannel-backend` | string | Sets the encapsulation technology that will allow inter-node traffic to work. It matches part of the Backend parameter in the subnet&backend configuration |
+| `--flannel-ipv6-masq` | bool | Enables masquerading traffic for IPv6. Note that IPv4 traffic is always masqueraded. It matches the ipv6 part of the `--ip-masq` flag |
+| `--flannel-external-ip` | bool | Enables using node external IP addresses for flannel traffic. It sort of matches on `--public-ip` and `--public-ipv6` flag. In this case, k3s selects the external-ips for those flags |
+
+### Design suggestion
+
+The suggestion is to include a string slice flag that would consolidate all flannel **bool** server flags into just one. Benefits:
+* Simplify user's life by having fewer flags
+* Use it for potential new flannel toggles
+
+## Proposal
+
+Starting in v1.26, introduce a new `flannel-opts` flag that includes flannel server options. The redundant flags are deprecated and removed in a few releases.
+
+We could reduce it to 2:
+
+* flannel-backend (string)
+* flannel-opts ([]string)
+
+Flannel-opts would have the following values:
```

Comment:
I would suggest making this `--flannel-opt`, but accepting either multiple comma-separated values, or being repeated multiple times with the values all being collected. This follow the convention of other flags such as `--kubelet-arg` - the flag name is singular (opt, arg, and so on).

Reply:
On the naming, that was my thought as well. I have included your thought on being able to repeat the arg multiple times.